src/extension_commands.ml

```diff
@@ -534,6 +534,193 @@ module Copy_type_under_cursor = struct
+        ?validationMessage:(if empty_result then validationMessage else None)
+        ~prompt:
+          "Perform a search by type request by providing a type signature to \
+           look for"
+        ()
+    in
+    let _disposable =
+      InputBox.onDidAccept
+        input_box
+        ~listener:(fun () ->
+          let query = InputBox.value input_box in
+          InputBox.set_busy input_box true;
+          InputBox.set_enabled input_box false;
+          resolve query)
+        ()
+    in
+    let _disposable =
+      InputBox.onDidChangeValue
+        input_box
+        ~listener:(fun _ -> InputBox.set_validationMessage input_box None)
+        ()
+    in
+    let _disposable =
+      InputBox.onDidHide input_box ~listener:(fun _ -> resolve None) ()
+    in
+    InputBox.show input_box
+
+  let get_search_results ~query ~limit ~with_doc ~position text_editor client =
+    let doc = TextEditor.document text_editor in
+    let uri = TextDocument.uri doc in
+    Custom_requests.(
+      send_request
+        client
+        Type_search.request
+        (Type_search.make ~uri ~position ~limit ~query ~with_doc ()))
+
+  let rec display_search_results query results text_editor position client =
+    let quickPickItems =
+      List.map
+        results
+        ~f:(fun (res : Custom_requests.Type_search.type_search_result) ->
+          QuickPickItem.create
+            ~label:res.name
+            ~description:res.typ
+            ~detail:(Option.value ~default:"" res.doc)
+            ())
+    in
+    let module QuickPick = Vscode.QuickPick.Make (QuickPickItem) in
+    let quickPick =
+      QuickPick.set
+        (Window.createQuickPick (module QuickPickItem) ())
+        ~title:"Type/Polarity Search Results"
+        ~activeItems:[]
+        ~busy:false
+        ~enabled:true
+        ~placeholder:"Select an item to insert it to the editor"
+        ~selectedItems:[]
+        ~ignoreFocusOut:true
+        ~items:quickPickItems
+        ~buttons:[ Window.quickInputButtonBack ]
+        ()
+    in
+    let _disposable =
+      QuickPick.onDidTriggerButton
+        quickPick
+        ~listener:(fun _ ->
+          let _ = handle_search ~previous_query:query text_editor client in
+          ())
+        ()
+    in
+    let _disposable =
+      QuickPick.onDidAccept
+        quickPick
+        ~listener:(fun () ->
+          match QuickPick.selectedItems quickPick with
+          | Some (item :: _) ->
+            let value = QuickPickItem.label item in
+            let _ =
+              Vscode.TextEditor.edit
+                text_editor
+                ~callback:(fun ~editBuilder ->
+                  Vscode.TextEditorEdit.insert
+                    editBuilder
+                    ~location:position
+                    ~value)
+                ()
+            in
+            QuickPick.hide quickPick
+          | _ ->
+            display_search_results query results text_editor position client)
```

Comment:
We could just do nothing here (`()`) and it will keep the results open. I also guess this case should never happen ?